File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/hybrid/HMPseudoState.java
package ru.nstu.isma.hsm.hybrid;

import ru.nstu.isma.hsm.exp.HMExpression;

import java.io.Serializable;
import java.util.UUID;

/**
 * Created by Bessonov Alex
 * on 08.02.2015.
 */
public class HMPseudoState extends HMState implements Serializable {

    protected HMExpression condition;

    public HMPseudoState() {
        super(UUID.randomUUID().toString());
    }

    public HMExpression getCondition() {
        return condition;
    }

    public void setCondition(HMExpression condition) {
        this.condition = condition;
    }
}

File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/hybrid/HMPseudoState.kt
package ru.nstu.isma.hsm.hybrid

import ru.nstu.isma.hsm.exp.HMExpression
import java.io.Serializable
import java.util.*

/**
 * Created by Bessonov Alex
 * on 08.02.2015.
 */
class HMPseudoState : HMState(UUID.randomUUID().toString()), Serializable {
    var condition: HMExpression? = null
}